test/Microsoft.IdentityModel.JsonWebTokens.Tests/JsonWebTokenHandlerValidationParametersTests.cs

```diff
@@ -0,0 +1,240 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Security.Claims;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.IdentityModel.TestUtils;
+using Microsoft.IdentityModel.Tokens;
+using Xunit;
+
+namespace Microsoft.IdentityModel.JsonWebTokens.Tests
+{
+    public class JsonWebTokenHandlerValidationParametersTests
+    {
+        [Theory, MemberData(nameof(JsonWebTokenHandlerValidationParametersTestCases))]
+        public async Task ValidateTokenAsync(JsonWebTokenHandlerValidationParametersTheoryData theoryData)
+        {
+            var context = TestUtilities.WriteHeader($"{this}.ValidateTokenAsync", theoryData);
+
+            JsonWebTokenHandler jsonWebTokenHandler = new JsonWebTokenHandler();
+
+            string jwtString;
+
+            if (theoryData.TokenString != null)
+            {
+                jwtString = theoryData.TokenString;
+            }
+            else
+            {
+                SecurityTokenDescriptor securityTokenDescriptor = new SecurityTokenDescriptor
+                {
+                    Subject = theoryData.Subject,
+                    SigningCredentials = theoryData.SigningCredentials,
+                    EncryptingCredentials = theoryData.EncryptingCredentials,
+                    AdditionalHeaderClaims = theoryData.AdditionalHeaderParams,
+                    Audience = theoryData.Audience,
+                    Issuer = theoryData.Issuer,
+                };
+
+                jwtString = jsonWebTokenHandler.CreateToken(securityTokenDescriptor);
+            }
+
+            TokenValidationResult tokenValidationResult = await jsonWebTokenHandler.ValidateTokenAsync(jwtString, theoryData.TokenValidationParameters);
```

Comment:
nit: we could use better names, perhaps

tokenValidationParameterResult and ValidationParameterResult

It makes reading the rest of the method easier.

Reply:
Updated.